R/utils.R

```diff
@@ -326,3 +326,28 @@ matching_variables <- function(variable_filters, variables) {
     not_found = not_found
   )
 }
+
+#' Returns a list of dimensions for the input variables.
+#' For this function to return the true dimensions of the variables, 
+#' the indexed variables must be listed in ascending order of indices.
+variable_dims <- function(variable_names = NULL) {
+  if(is.null(variable_names)) {
+    dims <- NULL
+  } else {
+    dims <- list()
+    uniq_variable_names <- unique(gsub("\\[.*\\]", "", variable_names))
+    var_names <- gsub("\\]", "", variable_names)
+    for(var in uniq_variable_names) {
+      pattern <- paste0(var, "\\[")
+      var_indices <- var_names[grep(pattern, var_names)]
+      var_indices <- gsub(pattern, "", var_indices)
+      if(length(var_indices)) {
+        var_indices <- strsplit(var_indices[length(var_indices)], ",")[[1]]
+        dims[[var]] <- as.numeric(var_indices)
+      } else {
+        dims[[var]] <- 0
```

Comment:
This is consistent with RStan, but what do you think about using 1 instead of 0 for a scalar real? I guess the question is whether we want to differentiate in R between the Stan types 

```
real alpha;
real alpha[1];
```

and I'm not sure if we do or don't. What do you think?

Reply:
I guess the question is what will people use this for?

Reply:
In R I think there is no difference so I do agree with just going with 1. 

The only reason for sticking with 0 is if this would be a problem for any packages looking into switchinh to cmdstanr. I dont have a good feeling if it would be.

Reply:
Yeah I guess that's possible, but probably unlikely. Since CmdStanR is new I think we should do what makes the most sense, which I think for an R users would be `1`.